sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -0,0 +1,179 @@
+        HttpRequest request = new HttpRequest(HttpMethod.GET, pollingContext.getData(LOCATION));
+        Mono<HttpResponse> responseMono;
+        if (context == null) {
+            responseMono = httpPipeline.send(request);
+        } else {
+            responseMono = httpPipeline.send(request, context);
+        }
+        return responseMono.flatMap(res -> {
+            HttpHeader locationHeader = res.getHeaders().get(LOCATION);
+            if (locationHeader != null) {
+                pollingContext.setData(LOCATION, locationHeader.getValue());
+            }
+
+            LongRunningOperationStatus status;
+            if (res.getStatusCode() == 202) {
+                status = LongRunningOperationStatus.IN_PROGRESS;
+            } else if (res.getStatusCode() >= 200 && res.getStatusCode() <= 204) {
+                status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+            } else {
+                status = LongRunningOperationStatus.FAILED;
+            }
+
+            return BinaryData.fromFlux(res.getBody()).flatMap(binaryData -> {
+                pollingContext.setData(POLL_RESPONSE_BODY, binaryData.toString());
+                if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, pollResponseType.getJavaType())) {
+                    return (Mono<T>) Mono.just(binaryData);
+                } else {
+                    return binaryData.toObjectAsync(pollResponseType);
+                }
+            }).map(pollResponse -> {
+                String retryAfter = res.getHeaderValue(RETRY_AFTER);
+                if (retryAfter != null) {
+                    return new PollResponse<>(status, pollResponse,
+                        Duration.ofSeconds(Long.parseLong(retryAfter)));
+                } else {
+                    return new PollResponse<>(status, pollResponse);
+                }
+            });
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<U> getResult(PollingContext<T> pollingContext, TypeReference<U> resultType) {
+        if (pollingContext.getLatestResponse().getStatus() == LongRunningOperationStatus.FAILED) {
+            return Mono.error(new RuntimeException("Long running operation failed."));
+        } else if (pollingContext.getLatestResponse().getStatus() == LongRunningOperationStatus.USER_CANCELLED) {
+            return Mono.error(new RuntimeException("Long running operation canceled."));
+        }
+
+        String finalGetUrl;
+        String httpMethod = pollingContext.getData(HTTP_METHOD);
+        if ("PUT".equalsIgnoreCase(httpMethod) || "PATCH".equalsIgnoreCase(httpMethod)) {
+            finalGetUrl = pollingContext.getData(REQUEST_URL);
+        } else if ("POST".equalsIgnoreCase(httpMethod) && pollingContext.getData(LOCATION) != null) {
+            finalGetUrl = pollingContext.getData(LOCATION);
+        } else {
+            throw logger.logExceptionAsError(new RuntimeException("Cannot get final result"));
```

Comment:
Return Mono.error() instead in all async methods

Reply:
Fixed.